### PR TITLE
Add Object support to csnex

### PR DESCRIPTION
**Pull request by gitlarryf**

### CODE REVIEW

File: exec/csnex/Global.cs [hunk 1]
```diff
@@ -464,11 +484,209 @@ public void number__toString()
         }
 #endregion
 #region Object Functions
+        #region Object Getters
+        public void object__getArray()
+        {
+            Object obj = Exec.stack.Pop().Object;
+
+            List<Object> a = new List<Object>();
```
Comment (ghewgill) [inline exec/csnex/Global.cs:492]: There's no purpose in creating a new `List` here, only to throw it away in the next call. Either don't initialize this reference (if the compiler lets you) or initialize it to null.

File: exec/csnex/Global.cs [hunk 2]
```diff
@@ -464,11 +484,209 @@ public void number__toString()
...
+            }
+
+            Exec.stack.Push(new Cell(r));
+        }
+
+        public void object__getNumber()
+        {
+            Cell obj = Exec.stack.Pop();
+
+            Number n = new Number(0);
```
Comment (ghewgill) [inline exec/csnex/Global.cs:569]: No need to create a new number here.

File: exec/csnex/Object.cs [hunk 3]
```diff
@@ -15,30 +49,210 @@ public virtual string toLiteralString()
...
+        }
+
+        public override bool subscript(Object index, ref Object r)
+        {
+            Number i = new Number(0);
+            if (!index.getNumber(ref i)) {
+                throw new NeonRuntimeException("DynamicConversionException", "to Number");
+            }
+            int ii = Number.number_to_int32(i);
+            if (ii >= Value.Count) {
```
Comment (ghewgill) [inline exec/csnex/Object.cs:72]: What happens if `ii` is negative here?

File: exec/csnex/Object.cs [hunk 4]
```diff
@@ -15,30 +49,210 @@ public virtual string toLiteralString()
...
+        {
+            StringBuilder r = new StringBuilder("{");
+            bool first = true;
+            foreach (KeyValuePair<string, Object> e in Value) {
+                if (first) {
+                    first = false;
+                } else {
+                    r.Append(", ");
+                }
+                r.Append("\"");
```
Comment (ghewgill) [inline exec/csnex/Object.cs:199]: Don't manually quote this here, use your `toLiteralString` or whatever. What if the key value itself contains quotes?

**Review by ghewgill — CHANGES_REQUESTED** [review]

File: exec/csnex/Global.cs (hunk 1 above)
Comment (gitlarryf) [reply exec/csnex/Global.cs:492]: Fixed.

File: exec/csnex/Global.cs (hunk 2 above)
Comment (gitlarryf) [reply exec/csnex/Global.cs:569]: Fixed.

File: exec/csnex/Object.cs (hunk 3 above)
Comment (gitlarryf) [reply exec/csnex/Object.cs:72]: Added a test for negative values.

File: exec/csnex/Object.cs (hunk 4 above)
Comment (gitlarryf) [reply exec/csnex/Object.cs:199]: Calling `string.Quote()` now instead.

**Pull request merged**